### PR TITLE
Feature Sync server settings enhancement

### PR DESCRIPTION
**Pull request by kalisp**
Sites are now configured in System Settings (unique names).

Settings schema and defaults changed, so Settings for all projects should be opened and resaved to get rid off warning during start.

Providers carry information about modifiable items and scopes where they can be modified.

Added new methods for Local Settings to get this info:
- `get_configurable_items([EditableScopes])` - returns dictionary for all projects and all sites (use [2] or [EditableScopes.LOCAL] for Local Setting scope)
- `get_configurable_items_for_project(project: str, scope:[])` - only for specific project
- `get_configurable_items_for_site(project:str, site:str, scope:[])`

Returned values:
```
                {projectA: {
                    siteA : {
                        key:"root", label:"root", value:"valueFromSettings", type"text"
                    }
                }
```
Implementation isn't breaking Python 2 hosts.

!!! This PR is also fixing 'studio' <> 'any remote' site use case, limiting physical uploads/download only in case that any of the sites is user one.

### CODE REVIEW

**Comment by kalisp** [comment]
Bit cleaned up return types:
**If you want to get all configurable properties for all sites (incl. defaults) for project:**
**get_local_settings_schema_for_project(PROJECT_NAME)**  (PROJECT_NAME == None >> returns default project
returns:
```
{
    siteA : [
        {
            key:"root", label:"root",
            "value":"{'work': 'c:/projects'}", "type": "dict"
        },
        {
          key:"....
        }
    ]
}
```

**If you want to get all configurable sites properties for specific site of specific project:**
**get_local_settings_schema_for_site(PROJECT_NAME, SITE_NAME)**
Returns:
```
[
    {
        key:"root", label:"root",
        "value":"{'work': 'c:/projects'}", "type": "text"
    }, ...
]
```

I am happy to change it to any defined format.

**Comment by kalisp** [comment]
Slightly changed format:
```
            {
                siteA : [
                    {
                        key:"root", label:"root",
                        "type": "dict",
                        "children":[
                            { "key": "work",
                              "type": "text",
                              "value": "c:/projects"}
                        ]
                    },
                    {
                        key:"credentials_url", label:"Credentials url",
                        "value":"'c:/projects/cred.json'", "type": "text",
                        "namespace": "{project_setting}/global/sync_server/
                                     sites"
                    }
                ]
            }
```

**Review by mkolar — APPROVED** [review]
functionaly seems ok. Let's wait on @iLLiCiTiT comments

File: openpype/modules/sync_server/providers/gdrive.py [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
-from googleapiclient.discovery import build
-import google.oauth2.service_account as service_account
-from googleapiclient import errors
-from .abstract_provider import AbstractProvider
-from googleapiclient.http import MediaFileUpload, MediaIoBaseDownload
+import time
+import sys
+from setuptools.extern import six
+import platform
+from json.decoder import JSONDecodeError
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply openpype/modules/sync_server/providers/gdrive.py:7]: `JSONDecodeError` not used and is not available in Python 2 there is only `ValueError`.

File: openpype/modules/sync_server/sync_server_module.py [hunk 2]
```diff
@@ -402,6 +396,200 @@ def get_remote_site(self, project_name):
...
+
+            items = self.get_configurable_items_for_project(project_name,
+                                                            scope)
+            editable.update(items)
+
+        return editable
+
+    def get_local_settings_schema_for_project(self, project_name):
+        """Wrapper for Local settings"""
+        return self.get_configurable_items(project_name, EditableScopes.LOCAL)
```
Comment (iLLiCiTiT) [inline openpype/modules/sync_server/sync_server_module.py:447]: Probably should be `get_configurable_items_for_project`

**Comment by iLLiCiTiT** [comment]
1. Configuable items are using settings with applied local settings to fill `"value"` in returned schema. But I need value withtou local settings to be able show user what it will be if he won't set it.

2. Value for specific project returns value from default project but should return value for the project.

File: openpype/modules/sync_server/sync_server_module.py (hunk 2 above)
Comment (kalisp) [reply openpype/modules/sync_server/sync_server_module.py]: Yes, fixed.

**Comment by iLLiCiTiT** [comment]
It seems that `"sudio"` site has right items but what should be `"local"` is called as workstation ID. BTW `gdrive` returned nothing, not sure it should.
```
"adamant-cobalt-boobook": [
        {
            "key": "root",
            "label": "Roots",
            "type": "dict",
            "children": []
        }
    ]
```

**Review by iLLiCiTiT — APPROVED** [review]
I think it does what it should do (for current purposes). Changed implementation it's not used as should because that require changes in local settings implementation but Anatomy paths should be updated as before.

**Pull request merged**